src/util.ts

```diff
@@ -59,6 +67,56 @@ export function toJose({ r, s, recoveryParam }: EcdsaSignature, recoverable?: bo
   return bytesToBase64url(jose)
 }
 
+export function fromJose(signature: string): { r: string; s: string; recoveryParam: number } {
+  const signatureBytes: Uint8Array = base64ToBytes(signature)
+  if (signatureBytes.length < 64 || signatureBytes.length > 65) {
+    throw new TypeError(
+      `Wrong size for signature. Expected 64 or 65 bytes, but got ${signatureBytes.length}`
+    )
+  }
+  const r = bytesToHex(signatureBytes.slice(0, 32))
+  const s = bytesToHex(signatureBytes.slice(32, 64))
+  const recoveryParam = signatureBytes.length === 65 ? signatureBytes[64] : undefined
+  return { r, s, recoveryParam }
+}
+
 export function toSealed(ciphertext: string, tag: string): Uint8Array {
   return u8a.concat([base64urlToBytes(ciphertext), base64urlToBytes(tag)])
 }
+
+const hexMatcher = /^(0x)?([a-fA-F0-9]{64}|[a-fA-F0-9]{128})$/
+const base58Matcher = /^([1-9A-HJ-NP-Za-km-z]{44}|[1-9A-HJ-NP-Za-km-z]{88})$/
+const base64Matcher = /^([0-9a-zA-Z=\-_\+\/]{43}|[0-9a-zA-Z=\-_\+\/]{86})(={0,2})$/
+
+/**
+ * Parses a private key and returns the Uint8Array representation.
+ * This method uses an heuristic to determine the key encoding to then be able to parse it into 32 or 64 bytes.
+ *
+ * @param input a 32 or 64 byte key presented either as a Uint8Array or as a hex, base64, or base58btc encoded string
+ * 
+ * @throws TypeError('Invalid private key format') if the key doesn't match any of the accepted formats or length
+ */
+export function parseKey(input: string | Uint8Array): Uint8Array {
+  if (typeof input === 'string') {
+    if (hexMatcher.test(input)) {
+      const parts = hexMatcher.exec(input)
+      return hexToBytes(parts[2])
+    } else if (base58Matcher.test(input)) {
+      return u8a.fromString(input, 'base58btc')
+    } else if (base64Matcher.test(input)) {
+      const inputBase64Url = input.replace(/\+/g, '-').replace(/\//g, '_').replace(/=/g, '')
```

Comment:
We can use `base64ToBytes` here, do deduplicate code, correct?

Reply:
yes!
great catch

Reply:
fixed